src/scripts/main.js

```diff
@@ -1,3 +1,45 @@
 'use strict';
 
-// write code here
+const list = document.querySelectorAll('ul li');
+
+function sortList() {
+  const listItems = [...list];
+
+  const numeric = () => {
+    for (const item of listItems) {
+      item.dataset.salary = Number(
+        item.dataset.salary.replace(/[^0-9.-]+/g, ''),
+      );
+    }
+  };
+
+  numeric();
+
+  listItems.sort((a, b) => b.dataset.salary - a.dataset.salary);
```

Comment:
The sorting function assumes that all 'salary' data attributes are valid numbers after the 'numeric' function processes them. Ensure that all list items have a 'data-salary' attribute to avoid potential issues.